src/components/radio/radio.spec.ts

```diff
@@ -1,370 +1,398 @@
+      expect(groupInstance.selected).toBe(radioInstances[1]);
+      expect(radioInstances[0].checked).toBe(false);
+      expect(radioInstances[1].checked).toBe(true);
     });
 
-    it('should be checked after input change event', (done: () => void) => {
-      builder
-        .overrideTemplate(TestApp, '<md-radio-button></md-radio-button>')
-        .createAsync(TestApp)
-        .then(fixture => {
-          let button = fixture.debugElement.query(By.css('md-radio-button'));
-          let input = button.query(By.css('input'));
-
-          fixture.detectChanges();
-          expect(button.componentInstance.checked).toBe(false);
-
-          let event = createEvent('change');
-          input.nativeElement.dispatchEvent(event);
-          expect(button.componentInstance.checked).toBe(true);
-        }).then(done);
+    it('should check a radio upon interaction with the underlying native radio button', () => {
+      let nativeRadioInput = <HTMLElement> radioNativeElements[0].querySelector('input');
+
+      nativeRadioInput.click();
+      fixture.detectChanges();
+
+      expect(radioInstances[0].checked).toBe(true);
+      expect(groupInstance.value).toBe('fire');
+      expect(groupInstance.selected).toBe(radioInstances[0]);
     });
 
-    it('should emit event when checked', (done: () => void) => {
-      builder
-        .overrideTemplate(TestApp, '<md-radio-button></md-radio-button>')
-        .createAsync(TestApp)
-        .then(fixture => {
-          fakeAsync(function() {
-            let button = fixture.debugElement.query(By.css('md-radio-button'));
-            let changeEvent: MdRadioChange = null;
-            button.componentInstance.change.subscribe((evt: MdRadioChange) => {
-              changeEvent = evt;
-            });
-            button.componentInstance.checked = true;
-            fixture.detectChanges();
-            tick();
-
-            expect(changeEvent).not.toBe(null);
-            expect(changeEvent.source).toBe(button.componentInstance);
-          });
-        }).then(done);
+    it('should emit a change event from radio buttons', () => {
+      expect(radioInstances[0].checked).toBe(false);
+
+      // TODO(jelbourn): this *should* work with async(), but fixture.whenStable currently doesn't
+      // know to look at pending macro tasks.
+      // See https://github.com/angular/angular/issues/8389
+      // As a short-term solution, use a promise (which jasmine knows how to understand).
+      let promiseCompleter = new PromiseCompleter();
+      let changeCount = 0;
+      radioInstances[0].change.subscribe(() => {
+        changeCount++;
+
+        // One event for checked, one for unchecked.
+        if (changeCount == 2) {
+          promiseCompleter.resolve();
+        }
+      });
+
+      radioInstances[0].checked = true;
+      fixture.detectChanges();
+
+      radioInstances[0].checked = false;
+      fixture.detectChanges();
+
+      return promiseCompleter.promise;
```

Comment:
Couldn't this be done more easily by using `spyOn`?  It seems needlessly confusing the way it's written.